processing/bloom_handler.go

```diff
@@ -188,29 +192,57 @@ func (a *BloomHandler) Reload() error {
 // Consume processes an Entry, emitting alerts if there is a match
 func (a *BloomHandler) Consume(e *types.Entry) error {
 	if e.EventType == "http" {
-		var checkStr string
+		var fullURL string
 		a.Lock()
+		// check HTTP host first: foo.bar.de
+		if a.IocBloom.Check([]byte(e.HTTPHost)) {
+			n := MakeAlertEntryForHit(*e, "http-host", a.AlertPrefix, e.HTTPHost)
+			a.DatabaseEventChan <- n
+			a.ForwardHandler.Consume(&n)
+		}
+		// we sometimes see full 'URLs' in the corresponding EVE field when
+		// observing requests via proxies. In this case there is no need to
+		// canonicalize the URL, it is already qualified.
 		if strings.Contains(e.HTTPUrl, "://") {
-			checkStr = e.HTTPUrl
+			fullURL = e.HTTPUrl
 		} else {
-			checkStr = "http://" + e.HTTPHost + e.HTTPUrl
+			// in all other cases, we need to create a full URL from the components
+			fullURL = "http://" + e.HTTPHost + e.HTTPUrl
+		}
+		// we now should have a full URL regardless of where it came from:
+		// http://foo.bar.de:123/baz
+		u, err := url.Parse(fullURL)
+		if err != nil {
+			log.Warnf("could not parse URL '%s': %s", fullURL, err.Error())
+			a.Unlock()
+			return nil
 		}
-		if a.IocBloom.Check([]byte(checkStr)) {
-			n := MakeAlertEntryForHit(*e, "http-url", a.AlertPrefix)
+
+		// http://foo.bar.de:123/baz
+		if a.IocBloom.Check([]byte(fullURL)) {
+			n := MakeAlertEntryForHit(*e, "http-url", a.AlertPrefix, fullURL)
 			a.DatabaseEventChan <- n
 			a.ForwardHandler.Consume(&n)
-		}
-		if a.IocBloom.Check([]byte(e.HTTPHost)) {
-			n := MakeAlertEntryForHit(*e, "http-host", a.AlertPrefix)
+		} else
+		// foo.bar.de:123/baz
+		if a.IocBloom.Check([]byte(fmt.Sprintf("%s%s", u.Host, u.Path))) {
+			n := MakeAlertEntryForHit(*e, "http-url", a.AlertPrefix, fmt.Sprintf("%s%s", u.Host, u.Path))
+			a.DatabaseEventChan <- n
+			a.ForwardHandler.Consume(&n)
+		} else
+		// /baz
```

Comment:
I wonder how the false-positive rate of this particular match will be.

Reply:
Depends on how specific the upstream IoC is written of course. Drop in some `/index.php` or even `/` and it's not going to be good obviously. If you think that this is too sharp a double-edged sword, I'd be happy to make it optional.

Reply:
Let's go with it for now and see how it works out in production.